ruby/lib/ci/queue/redis/supervisor.rb

```diff
@@ -21,17 +21,33 @@ def wait_for_workers
 
           yield if block_given?
 
-          time_left = config.timeout
-          until exhausted? || time_left <= 0 || max_test_failed?
-            sleep 1
+          time_left = config.report_timeout
+          time_left_with_no_workers = config.queue_init_timeout
```

Comment:
I wonder if you need to set some minimum default in case somebody set this really low (e.g. to 1 like what we did in Core)?

cc @casperisfine

Reply:
Yeah, definitely shouldn't be set too low.